hack/validate.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+set -ex
+
+if [[ -z "$1" ]]; then
+    echo "run me with either 'codegen' or 'modcheck' argument"
+    exit 1
+fi
+
+#############
+# FUNCTIONS #
+#############
+function validate_codegen(){
+    for file in $(git status --porcelain); do
+        if [[ "$file" =~ "zz" ]]; then
+            echo "found codegen files!"
+            echo "please run 'make codegen' and update your PR"
+            exit 1
+        fi
+    done
+}
+
+function validate_modcheck(){
+    if ! git diff-index --quiet HEAD; then
```

Comment:
The purpose of this check was just to see if any files were modified. Seems like we could use the same check for both codegen and modcheck, no need for separate functions.